### ISSUE TITLE
Command is invalid when running test prepare command through Azure DevOps task

### ISSUE THREAD

**Issue by rygo-msft** [microsoft/appcenter-cli#696]
Starting last week (11/7 4:00 AM EST) our AppCenterTest runs have been failing to submit from Azure DevOps.

When running in the vs2017-win2016 hosted agent image using the AppCenterTest task, the prepare step has begun failing. It does not appear to be an issue with the task as the command sent to the CLI is identical to what successful jobs were sending.

```
- task: AppCenterTest@1
  name: RunAppCenterTests
  displayName: 'Test with Visual Studio App Center'
  inputs:
    espressoBuildDirectory: ${{ parameters.appBuildDir }}
    appFile: ${{ parameters.appFile }}
    artifactsDirectory: AppCenterTest
    frameworkOption: espresso
    serverEndpoint: 9015e276-e51e-4284-bcf0-6b56a5517017 #MAM functional test connection
    appSlug: ${{ parameters.appCenterAppId }}
    devices: $(DeviceSelector.Devices)
    runOptions: ${{ parameters.runOptions }}
```

```
C:\windows\system32\cmd.exe /D /S /C "D:\a\_tasks\AppCenterTest_ad5cd22a-be4e-48bb-adce-181a32432da5\1.152.1\node_modules\.bin\appcenter.cmd test prepare espresso --artifacts-dir D:\a\1\s\AppCenterTest --build-dir D:\a\1\a\MAM\AppClient.Test.Functional --quiet"
Error: Command test prepare espresso --artifacts-dir D:\a\1\s\AppCenterTest --build-dir D:\a\1\a\MAM\AppClient.Test.Functional --quiet is invalid
```

**Comment by rygo-msft** [comment]
I am also seeing people reporting similar issues in #688

**Comment by jpc0001** [comment]
I'm seeing the same thing on a windows-2019 agent

**Comment by joabalea** [comment]
It seems that you are running a node version that is not compatible with the CLI node version.
From my tests, it seems that `windows-latest` is using node (v12.13.0) whereas for now CLI requires node 10.x.

You could try adding the following task before the App Center task to downgrade the agent node version:

`
- task: UseNode@1
  inputs:
    version: '10.15.1'
`

**Comment by rygo-msft** [comment]
@joabalea thank you for the workaround. Is the issue that AppCenterTest task is not setting its required Node version?
When can we expect a fix for this so that we can remove the work around?

**Comment by MouthOfMadness** [comment]
I'm seeing this error with the UITest framework as well.

**Comment by Azhimkulov** [comment]
I can't understand how to add:
`- task: UseNode@1
  inputs:
    version: '10.15.1'`

Can somebody explain to me how to do this?

**Comment by JorisAbalea** [comment]
@Azhimkulov This should be a task for your AzureDevOps pipeline. If you are using .yaml file, you can just copy past it before the App Center test task (`- task: AppCenterTest@1 ...` ). If you are using the UI pipeline view, you should be able to search for a task called `Use Node.js ecosystem` and set it to `10.15.1` (also before your App Center test task):

<img width="369" alt="Screen Shot 2019-11-14 at 15 36 58" src="https://user-images.githubusercontent.com/20119274/68832735-1490a780-06f5-11ea-85f0-41c3c44d80ab.png">

**Comment by JorisAbalea** [comment]
@rygo-msft We are working on prioritizing this and it should be shipped with the next version of the task.

**Comment by joabalea** [comment]
@rygo-msft We are working on prioritizing this and it should be shipped with the next version of the task.

**Comment by joabalea** [comment]
@Azhimkulov This should be a task for your AzureDevOps pipeline. If you are using .yaml file, you can just copy past it before the App Center test task (- task: `AppCenterTest@1 ... `). If you are using the UI pipeline view, you should be able to search for a task called Use `Node.js ecosystem` and set it to `10.15.1 `(also before your App Center test task):

<img width="369" alt="Screen Shot 2019-11-14 at 15 36 58" src="https://user-images.githubusercontent.com/39932946/68833098-fd9e8500-06f5-11ea-968e-3e6a78d2fc57.png">

**Comment by svaldetero** [comment]
> @rygo-msft We are working on prioritizing this and it should be shipped with the next version of the task.

It's been almost a year, has this been fixed?

**Comment by danwaters** [comment]
Still broken for me too. By the way, the "use node" extension no longer shows up in search at the time of this posting; you need to search for *Node.js tool installer*

**Comment by MouthOfMadness** [comment]
I've got mine working in the release pipeline (and I have one working in yaml file), but I'm going to post mine as the release pipeline.

This is what my pipeline looks like, the difference in the way this used to work and the way I have it working, is you now have to run the task twice, once to prepare the test, then to run the test.  
![image](https://user-images.githubusercontent.com/29233402/100635744-7fa81e80-32f6-11eb-8ec6-17b2b1e2f776.png)

I came to believe (without proof) that the prepare test generates the manifest file used by the test as this always seemed to be my point of failure (in this folder $(Build.ArtifactStagingDirectory)/AppCenterTest).
![image](https://user-images.githubusercontent.com/29233402/100636144-00671a80-32f7-11eb-8178-75ff986c670e.png)

This is what the yaml looks like for each task in the release pipeline (paths, tools, test-assembly are all relative to your project):

steps:
- task: UseNode@1
  displayName: 'Use Node 6.x'
  inputs:
    version: 10.15.1

- task: AppCenterTest@1
  displayName: 'Prepare Test with Visual Studio App Center'
  inputs:
    appFile: '$(path)/andriod.apk'
    frameworkOption: uitest
    uiTestBuildDirectory: '$(path)/test-assembly'
    uiTestToolsDirectory: '$(path)/tools'
    serverEndpoint: MyAppCenterUser
    appSlug: 'mySlug'
    devices: 'myDevices'
    skipWaitingForResults: true

- task: AppCenterTest@1
  displayName: 'Test with Visual Studio App Center'
  inputs:
    appFile: '$(path)/android.apk'
    prepareTests: false
    serverEndpoint: MyAppCenterUser
    appSlug: 'mySlug'
    devices: 'myDevices'
    skipWaitingForResults: true
